README.md

```diff
@@ -115,61 +125,187 @@ add your metadata as an entry into the `NON_PUDL_SOURCES` dictionary in
+method takes a URL, and a `regex` pattern, and it will find all hyperlinks matching the
+pattern on the page pointed to by the URL. This is useful if there's a page containing
+links to a series of data resources that have somewhat structured names.
+* Calls an API to identify download queries for each partition of the data, and yields
+awaitables downloading each partition of the data from the API (see `archivers.eia.epacems.py`).
+
+In the example above, `get_resources` is defined as follows:
+```py
+async def get_resources(self) -> ArchiveAwaitable:
+  """Download EIA-860 resources."""
+  link_pattern = re.compile(r"eia860(\d{4})(ER)*.zip")
+  for link in await self.get_hyperlinks(BASE_URL, link_pattern):
+      matches = link_pattern.search(link)
+      if not matches:
+          continue
+      year = int(matches.group(1))
+      if self.valid_year(year):
+          yield self.get_year_resource(link, year)
+```
+In this case, we know that Form 860 data is on a webpage (`BASE_URL`) containing a
+series of download links, and that the links to the data we want follow a general pattern:
+they are called `eia860{year}.zip` or `eia860{year}ER.zip`. We search through all the
+links in `BASEURL` to find links that match this pattern. For each matching link, we
+extract the year from the file name and pass both the link and the year to the
+`get_year_resource()` method.
+
+> [!TIP]
+> `self.valid_year()` is an optional method that allows us to easily run the archiver on
+> only a year or two of data, for datasets partitioned by year. Though optional, it helps
+> to speed up testing of the data. The method expects a year and returns a boolean
+> indicating whether or not the year is valid.
+
+#### Getting each individual resource
+
+In the example above, we define a second async method. This method downloads a single file per partition:
+```py
+async def get_year_resource(self, link: str, year: int) -> ResourceInfo:
+  """Download zip file."""
+  # Append hyperlink to base URL to get URL of file
+  url = f"{BASE_URL}/{link}"
+  download_path = self.download_directory / f"eia860-{year}.zip"
+  await self.download_zipfile(url, download_path)
+
+  return ResourceInfo(local_path=download_path, partitions={"year": year})
+```
+This method should handle the following steps:
+* identify the specific download link for the file(s) in the partition
+* construct the name of the downloaded file. We rename files to match the format
+`datasource-year.ext` - e.g. `eia860-1990.zip`.
+* construct the path to the downloaded file, using `self.download_directory`: this is
+a temporary directory created and manged by the base class that is used as a staging area
+for downloading data before uploading it to Zenodo. This temporary directory will be
+automatically removed once the data has been uploaded.
+* return `ResourceInfo`, where `local_path` is the path to the downloaded file and
+`partitions` is a dictionary specifying the partition(s) of the dataset.
```

Comment:
clarifying question: we are not returning the downloaded file itself because we have already put it in `download_path` and this `ResourceInfo` is the post-download info of where to grab to file later once all the files have been downloaded?